cdv/cmds/sim_utils.py

```diff
@@ -0,0 +1,525 @@
+    bt = BlockTools(
+        test_constants,
+        root_path,
+        automated_testing=False,
+        plot_dir=config["simulator"].get("plot_directory", "plots"),
+        keychain=Keychain(),
+    )
+    await bt.setup_keys(fingerprint=fingerprint, reward_ph=farming_puzzle_hash)
+    starting_time = time()
+    await bt.setup_plots(num_og_plots=PLOTS, num_pool_plots=0, num_non_keychain_plots=0, plot_size=PLOT_SIZE)
+    print(f"{'New plots generated.' if time() - starting_time > 5 else 'Using Existing Plots'}\n")
+
+
+async def get_current_height(node_client: SimulatorFullNodeRpcClient, _config: Dict[str, Any]) -> int:
+    # this is a helper function to get the current height of the simulator, using execute_with_simulator
+    num_blocks = len(await node_client.get_all_blocks())
+    return num_blocks
+
+
+async def async_config_wizard(
+    root_path: Path,
+    fingerprint: Optional[int],
+    farming_address: Optional[str],
+    plot_directory: Optional[str],
+    mnemonic_string: Optional[str],
+    auto_farm: Optional[bool],
+    docker_mode: bool,
+) -> None:
+    # either return passed through fingerprint or get one
+    fingerprint = select_fingerprint(fingerprint, mnemonic_string, docker_mode)
+    if fingerprint is None:
+        # user cancelled wizard
+        return
+    # create chia directory & get config.
+    print("Creating chia directory & config...")
+    config = create_chia_directory(root_path, fingerprint, farming_address, plot_directory, auto_farm, docker_mode)
+    # Pre-generate plots by running block_tools init functions.
+    print("Please Wait, Generating plots...")
+    print("This may take up to a minute if you are on a slow machine")
+
+    await generate_plots(config, root_path, fingerprint)
+    # final messages
+    farming_address = config["simulator"]["farming_address"]
+    print(f"\nFarming & Prefarm reward address: {farming_address}\n")
+    print("Configuration Wizard Complete.")
+    print("Starting Simulator now...\n\n")
+    await start_async(root_path, ("simulator",), False)
+    # now we create make sure the simulator has a genesis block
+    print("Please wait, generating genesis block.")
+    while True:
+        try:
+            num_blocks = await execute_with_simulator(None, root_path, get_current_height, False)
+        except ClientConnectorError:
+            await asyncio.sleep(0.25)
+        else:
+            if num_blocks == 0:
+                await execute_with_simulator(None, root_path, farm_blocks, False, 1, True, farming_address)
+                print("Genesis block generated, exiting.")
+            else:
+                print("Genesis block already exists, exiting.")
+            break
+
+
+def print_coin_record(
+    name: str,
+    address_prefix: str,
+    mojo_per_unit: int,
+    coin_record: CoinRecord,
+) -> None:
+    from datetime import datetime
+
+    chia_amount = Decimal(int(coin_record.coin.amount)) / mojo_per_unit

```

Comment:
Not a big fan of this, you're likely only looking at coin records if you're doing some wallet or chialisp development and you're likely going to need to convert to mojos anyway.  Same with the address conversion.  I have no problems with throwing them as additional information, but not replacing puzzle hash and amount.

Reply:
the entire point of this simulator is for wallet or chialisp development.

Reply:
That's my point, wallet/chialisp development isn't going to ever want the decmials only the mojo amounts

Reply:
> Not a big fan of this, you're likely only looking at coin records if you're doing some wallet or chialisp development and you're likely going to need to convert to mojos anyway. Same with the address conversion. I have no problems with throwing them as additional information, but not replacing puzzle hash and amount.
Oh i 100% over thought that, i'll fix that shortly.